x/emissions/keeper/msgserver/msg_server_worker_payload.go

```diff
@@ -58,20 +56,14 @@ func (ms msgServer) InsertWorkerPayload(ctx context.Context, msg *types.MsgInser
 		)
 	}
 
-	hasEnoughBal, fee, err := ms.CheckBalanceForSendingDataFee(ctx, msg.Sender)
+	// Before creating topic, transfer fee amount from creator to ecosystem bucket
+	params, err := ms.k.GetParams(ctx)
 	if err != nil {
-		return nil, errorsmod.Wrapf(err,
-			"Error checking balance for sender: %v", &msg.Sender)
-	}
-	if !hasEnoughBal {
-		return nil, types.ErrDataSenderNotEnoughDenom
+		return nil, errorsmod.Wrapf(err, "Error getting params for sender: %v", &msg.Sender)
 	}
-
-	// Before creating topic, transfer fee amount from creator to ecosystem bucket
-	err = ms.k.SendCoinsFromAccountToModule(ctx, msg.Sender, mintTypes.EcosystemModuleName, sdk.NewCoins(fee))
+	err = sendEffectiveRevenueActivateTopicIfWeightSufficient(ctx, ms, msg.Sender, topicId, params.DataSendingFee, "insert worker payload")
```

Comment:
Is there a reason to add topic fee revenue when inserting worker/reputation payload?

Reply:
Responding for completeness here as well:

whenever we pay into the ecosystem bucket or add stake, both should impact the topic weight.
for revenue to impact topic weight, we must update effective revenue
and any increase to topic weight should be coupled with possible reactivation.
we automatically inactivate but “manually” activate (per relevant tx)
=> need to capture all the moments when reactivation is possible
inactivation is done automatically already in EndBlock